### DISCUSSION TITLE
Delegate methods of aasm to Tramway Decorator and Tramway Form

### DISCUSSION THREAD

**Discussion by kalashnikovisme** [Purple-Magic/tramway#56] [Ideas]
Using Tramway for a commercial product, I faced an issue, that I had to do something like this every time and it became annoying :slightly_smiling_face: 

*app/models/user.rb*
```ruby
class User < ActiveRecord::Base
  aasm do
    state :unviewed
    state :approved
    state :declined
  end
end
```

*app/decorators/user_decorator.rb*
```ruby
class UserDecorator < Tramway::BaseDecorator
  delegate_attributes :unviewed?, :approved?, :declined?
end
```

I think that adding something integration with the popular gem should be ok.

What do you think about this helper to be added?

```ruby
class UserDecorator < Tramway::BaseDecorator
  delegate_aasm
end
```

**Comment by kalashnikovisme** [comment]
@haukot @tkarpesh @moshinaan

**Reply by haukot** [reply]
At this point it's look like you will want to delegate automatically more and more :) Maybe it's better to find what you *don't* want to delegate, and delegate all other things?

**Reply by kalashnikovisme** [reply]
I want to keep a thin decorator either but provide flexibility with it. In the case of `update` and `destroy` methods we discussed earlier, it would be changed to `behave_as_ar` method. So, by default decorator would be thin with the ability to have more functions :slightly_smiling_face:

**Reply by haukot** [reply]
Sure, I'm talking about another thing - in a typical Rails application, what attributes will you want to write manually? 
Maybe it'll be better to create one helper to delegate all methods *without* those.

Because current vector of development - add one helper for AR, another for aasm. Maybe next one for Discard, for PaperTrail, etc etc.
How many libs will be supported, for how long? Wouldn't it be too much work for not so core thing?

**Reply by kalashnikovisme** [reply]
The brilliance of open-source maintaining is that we are able to remove everything we want to :slightly_smiling_face: Only a bunch of time between removing announcement and removement matters :slightly_smiling_face: It depends on the count of active users :slightly_smiling_face:

**Reply by kalashnikovisme** [reply]
For now, we can use ar, aasm, etc. helpers for extending of Tramway Decorator. In case, we do not need them later, we will ask the community about it, and if nobody is against such a decision, we will just remove that :laughing:

**Reply by haukot** [reply]
Main question is not if it's easy or not to remove, but if it's worth it. Anyway, if you feel like it is the way - go for it :)
